File: src/lib/types/equipements.ts
// Types équipements — Phase 4

export interface CategorieModele {
  id_categorie: number;
  nom_categorie: string;
  description: string | null;
}

export interface ModeleEquipement {
  id_modele_equipement: number;
  nom_modele: string;
  description: string | null;
  id_categorie: number | null;
  nom_categorie: string | null;
  date_creation: string | null;
  date_modification: string | null;
  nb_champs: number;
  nb_familles: number;
}

export interface ChampModele {
  id_champ: number;
  id_modele_equipement: number;
  nom_champ: string;
  type_champ: "texte" | "nombre" | "date" | "booleen" | "liste";
  unite: string | null;
  est_obligatoire: number;
  ordre: number;
  valeurs_possibles: string | null;
  valeur_defaut: string | null;
  est_archive: number;
}

export interface ValeurChampEquipement {
  id_champ: number;
  nom_champ: string;
  type_champ: "texte" | "nombre" | "date" | "booleen" | "liste";
  unite: string | null;
  est_obligatoire: number;
  ordre: number;
  valeurs_possibles: string | null;
  valeur_defaut: string | null;
  est_archive: number;
  valeur: string | null;
}

export interface Domaine {
  id_domaine: number;
  nom_domaine: string;
  description: string | null;
  id_image: number | null;
}

export interface DomaineEquipListItem {
  id_domaine: number;
  nom_domaine: string;
  description: string | null;
  id_image: number | null;
  nb_familles: number;
  nb_equipements_inactifs: number;
  nb_equipements_total: number;
  nb_ot_en_retard: number;
  nb_ot_reouvert: number;
  nb_ot_en_cours: number;
  prochaine_date: string | null;
  jours_periodicite_min: number | null;
}

export interface Famille {
  id_famille: number;
  nom_famille: string;
  description: string | null;
  id_domaine: number;
  id_image: number | null;
  id_modele_equipement: number;
}

export interface FamilleEquipListItem {
  id_famille: number;
  nom_famille: string;
  description: string | null;
  id_image: number | null;
  nom_modele: string;
  nb_equipements: number;
  nb_equipements_inactifs: number;
  nb_ot_en_retard: number;
  nb_ot_reouvert: number;
  nb_ot_en_cours: number;
  prochaine_date: string | null;
  jours_periodicite_min: number | null;
}

export interface Equipement {
  id_equipement: number;
  nom_affichage: string;
  date_mise_en_service: string | null;
  date_fin_garantie: string | null;
  id_famille: number;
  id_local: number | null;
  est_actif: number;
  commentaires: string | null;
  id_image: number | null;
  date_creation: string | null;
  date_modification: string | null;
}

export interface EquipementListItem {
  id_equipement: number;
  nom_affichage: string;
  description: string | null;
  est_actif: number;
  id_image: number | null;
  nb_ot_en_retard: number;
  nb_ot_reouvert: number;
  nb_ot_en_cours: number;
  prochaine_date: string | null;
  jours_periodicite_min: number | null;
}

export interface EquipementSelectItem {
  id_equipement: number;
  nom_affichage: string;
}

File: src-tauri/src/models/equipements.rs
use serde::{Deserialize, Serialize};

// ── Domaines techniques ──

#[derive(Debug, Serialize, Deserialize)]
pub struct Domaine {
    pub id_domaine: i64,
    pub nom_domaine: String,
    pub description: Option<String>,
    pub id_image: Option<i64>,
}

#[derive(Debug, Deserialize)]
pub struct DomaineInput {
    pub nom_domaine: String,
    pub description: Option<String>,
    pub id_image: Option<i64>,
}

/// Version liste enrichie pour affichage en cartes (domaines équipements)
#[derive(Debug, Serialize, Deserialize)]
pub struct DomaineEquipListItem {
    pub id_domaine: i64,
    pub nom_domaine: String,
    pub description: Option<String>,
    pub id_image: Option<i64>,
    pub nb_familles: i64,
    pub nb_equipements_inactifs: i64,
    pub nb_equipements_total: i64,
    pub nb_ot_en_retard: i64,
    pub nb_ot_reouvert: i64,
    pub nb_ot_en_cours: i64,
    pub prochaine_date: Option<String>,
    pub jours_periodicite_min: Option<i64>,
}

// ── Familles d'équipements ──

#[derive(Debug, Serialize, Deserialize)]
pub struct Famille {
    pub id_famille: i64,
    pub nom_famille: String,
    pub description: Option<String>,
    pub id_domaine: i64,
    pub id_image: Option<i64>,
    pub id_modele_equipement: i64,
}

#[derive(Debug, Deserialize)]
pub struct FamilleInput {
    pub nom_famille: String,
    pub description: Option<String>,
    pub id_domaine: i64,
    pub id_image: Option<i64>,
    pub id_modele_equipement: i64,
}

/// Version liste enrichie pour affichage en cartes (familles équipements)
#[derive(Debug, Serialize, Deserialize)]
pub struct FamilleEquipListItem {
    pub id_famille: i64,
    pub nom_famille: String,
    pub description: Option<String>,
    pub id_image: Option<i64>,
    pub nom_modele: Option<String>,
    pub nb_equipements: i64,
    pub nb_equipements_inactifs: i64,
    pub nb_ot_en_retard: i64,
    pub nb_ot_reouvert: i64,
    pub nb_ot_en_cours: i64,
    pub prochaine_date: Option<String>,
    pub jours_periodicite_min: Option<i64>,
}

/// Version liste enrichie pour affichage en cartes (équipements)
#[derive(Debug, Serialize, Deserialize)]
pub struct EquipementListItem {
    pub id_equipement: i64,
    pub nom_affichage: String,
    pub description: Option<String>,
    pub est_actif: i64,
    pub id_image: Option<i64>,
    pub nb_ot_en_retard: i64,
    pub nb_ot_reouvert: i64,
    pub nb_ot_en_cours: i64,
    pub prochaine_date: Option<String>,
    pub jours_periodicite_min: Option<i64>,
}

/// Item léger pour les Select dropdown (id + nom)
#[derive(Debug, Serialize)]
pub struct EquipementSelectItem {
    pub id_equipement: i64,
    pub nom_affichage: String,
}

// ── Équipements ──

#[derive(Debug, Serialize, Deserialize)]
pub struct Equipement {
    pub id_equipement: i64,
    pub nom_affichage: String,
    pub date_mise_en_service: Option<String>,
    pub date_fin_garantie: Option<String>,
    pub id_famille: i64,
    pub id_local: Option<i64>,
    pub est_actif: i64,
    pub commentaires: Option<String>,
    pub id_image: Option<i64>,
    pub date_creation: Option<String>,
    pub date_modification: Option<String>,
}

#[derive(Debug, Deserialize)]
pub struct EquipementInput {
    pub nom_affichage: String,
    pub date_mise_en_service: Option<String>,
    pub date_fin_garantie: Option<String>,
    pub id_famille: i64,
    pub id_local: Option<i64>,
    pub est_actif: i64,
    pub commentaires: Option<String>,
    pub id_image: Option<i64>,
}
